src/generators/indexGenerator.ts

```diff
@@ -24,21 +24,82 @@ export function generateIndexFile(
       );
     }
     generateHLCIndex(clientDetails, indexFile);
-  } else {
+  } else if (!batch || batch?.length === 1) {
```

Comment:
Can you add a comment that this is the case where a single client is being generated?